AK/FixedCharBuffer.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Array.h>
+#include <AK/StringView.h>
+#include <AK/TypedTransfer.h>
+
+namespace AK {
+
+template<size_t Size>
+class FixedCharBuffer {
+public:
+    void store_characters(StringView characters)
+    {
+        m_storage.fill(0);
+        AK::TypedTransfer<u8>::copy(m_storage.data(), characters.bytes().data(), min(m_storage.size(), characters.bytes().size()));
+    }
+
+    StringView raw_view() const { return StringView(m_storage.data(), Size); }
+    StringView representable_view() const { return StringView(m_storage.data(), Size).trim_at_first_null(); }
```

Comment:
If the entire point of this class is to store nul terminated strings, surely we should be able to always know how long the string is? We shouldn't let callers use the unsafe pointers, and always direct them to nice APIs to copy an exact number of chars in, so that we can know how many bytes are in use at any time.

At they very least, any "raw" operations should never just return a pointer. we have writable span classes for a reason.

Reply:
> If the entire point of this class is to store nul terminated strings

This was never the point of the class - it can store a string with no null terminators if we need to (for example, if you can set a process name to a string with 32 chars, then the last one could be anything too).

However, knowing how much bytes we store currently could be useful so I will look into that.